### ISSUE TITLE
Enum type inferred as never when using Type.Mapped()

### ISSUE THREAD

**Issue by tchyra** [sinclairzx81/typebox#897]
Reproduction:

```ts
enum E { O1, O2 }

const T = Type.Object({ a: Type.Enum(E) });                     // TObject<{ a: TEnum<typeof E> }>
type T = Static<typeof T>;                                      // { a: E }

const U = Type.Mapped(Type.KeyOf(T), K => Type.Index(T, K));    // TObject<{ a: TUnion<[]> }>
type U = Static<typeof U>;                                      // { a: never; }
```

Thank you for your work on Typebox, it's proven quite useful :)

**Comment by sinclairzx81** [comment]
@tchyra Hi, thanks for reporting!

Have just pushed a update to fix this on `0.32.32`. You should be able to test the update at the following URL (just be sure to clear out your cache)

[TypeScript Link Here](https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzgFQJ5gKYBo4GUYCGMwAxnAL5wBmUEIcA5AAIDOwAdiQDYHBQD0MdBgBGEAB4MAUFIzsArvQCiiOAHkAjDjUAmCjJIR2LeMjgBeFMIB0akQCsMJGAAokBAFxXM1pQpAuSgCUFEEA3HCRUdExsfz8KHaOzgA87l7IfoopQpgQVHBKAHwURVK5GCgWeITEJDnC+ShFEbFt7R1t8aqehfpShsbwAKrVaD4AsgRgmAAmLuMY1gDSGKhqVAtBOMsWJYvWAJLssxjiCztB4dHdyElOMGlwvcjD7MBGKQDaALol5GUKnBRpZ8ERSA08gVhi1OnD4TFuuk4OwMAA3DBQCLkIA)

Will close up this issue for now, but if you run into any issues, feel free to post on this thread.
All the best!
S

**Comment by tchyra** [comment]
Updated in my project and it seems to work perfectly, thank you very much for fixing this so quickly!
